packages/babel-plugin/src/shared/stylex-position-try.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict
+ */
+
+import type { InjectableStyle, StyleXOptions } from './common-types';
+
+import createHash from './hash';
+import expandShorthands from './preprocess-rules/index';
+import generateLtr from './physical-rtl/generate-ltr';
+import generateRtl from './physical-rtl/generate-rtl';
+import transformValue from './utils/transform-value';
+import dashify from './utils/dashify';
+import {
+  objFromEntries,
+  objEntries,
+  objMapKeys,
+  objMap,
+  Pipe,
+} from './utils/object-utils';
+import { defaultOptions } from './utils/default-options';
+
+// Similar to `stylex.keyframes` but for position-try-fallbacks
+// Takes an object of positioning properties and returns a string after hashing it
+// The generated string must be prefixed with -- for anchor positioning
+export default function styleXPositionTry(
+  styles: { +[string]: string | number },
+  options: StyleXOptions = defaultOptions,
+): [string, InjectableStyle] {
+  const { classNamePrefix = 'x' } = options;
+  const expandedObject = Pipe.create(styles)
+    .pipe((styles) => expandPositionShorthands(styles, options))
```

Comment:
Why do shorthands need to be expanded here?

Reply:
This is a badly named function. It does the appropriate thing for each config option and doesn’t expand shorthands in either of the non-legacy config options.